lib/rewards/server/subscribers/rewards-subscriber.js

```diff
@@ -0,0 +1,211 @@
+const { sc, Logger } = require('@reldens/utils');
+const { Modifier } = require('@reldens/modifiers');
+
+class RewardsSubscriber
+{
+    static async giveRewards(playerSchema, targetObject, eventsManager)
+    {
+        let rewards = sc.get(targetObject, 'rewards', false);
+
+        let winningRewards = this.getWinningRewards(rewards);
+        if (0 === winningRewards.length) {
+            return false;
+        }
+
+        let repositories = this.getRepositories(targetObject);
+
+        for (let winningReward of winningRewards) {
+            let giveRewardResult = await this.giveReward(winningReward, repositories, playerSchema, async (item) => {
+                await eventsManager.emit('reldens.rewardDropped', {
```

Comment:
don't pass the event as callback, just add a condition for `if(true=== giveRewardsResult)` and execute the event here.

Reply:
should rewardDropped event only emit on item reward type? giveRewardsResult will be true even if a modifier's type or experience type reward is given.

Reply:
I could return from the "giveReward" method an object declaring which reward type was given and and then do the validation `if (item_type === true) => emit event`.